### DISCUSSION TITLE
Domotizes the Itron water meter with esp and cc1101 in ha for 10€ - Haade.fr

### DISCUSSION THREAD

**Discussion by giscus** [haade-administrator/haade.fr#31] [Comments]
# Domotizes the Itron water meter with esp and cc1101 in ha for 10€ - Haade.fr

Automate your itron water meter step by step equipped with a 433mhz everblu cyble sensor with arduino in homeassistant thanks to an esp8266/esp32 module and a TI cc1101 transmitter/receiver

https://haade.fr/en/blog/automate-water-meter-itron-everblu-cyble

**Comment by babyloon00** [comment]
Bonjour,
j'essaye de faire cette installation sur la même carte que vous (D1 Mini). 
J'utilise l'IDE 1,8,19 (j'ai essayé avec la 2 aussi, mes problème) sous OS X Monterey. 
ESP8266 Community V 3.1.2
Lorsque je compile, j'ai ce genre d'erreurs:

<code>
Users/damien/Downloads/watermeter2mqtt/cc1101.cpp:795:91: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  795 |   if (!receive_radian_frame(0x12, 150, rxBuffer, sizeof(rxBuffer))) echo_debug(debug_out, "TMO on REC\n");
      |                                                                                           ^~~~~~~~~~~~~~
/Users/damien/Downloads/watermeter2mqtt/cc1101.cpp:812:27: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
  812 |     echo_debug(debug_out, "TMO on REC\n");
      |                           ^~~~~~~~~~~~~~
/Users/damien/Downloads/watermeter2mqtt/watermeter2mqtt.ino:32:1: warning: ISO C++ forbids converting a string constant to 'char*' [-Wwrite-strings]
   32 | "{                    \
      | ^~~~~~~~~~~~~~~~~~~~~~~
   33 | \"liters\": %d,       \
      | ~~~~~~~~~~~~~~~~~~~~~~~
   34 | \"counter\" : %d,     \
      | ~~~~~~~~~~~~~~~~~~~~~~~
   35 | \"battery\" : %d,     \
      | ~~~~~~~~~~~~~~~~~~~~~~~
   36 | \"timestamp\" : \"%s\"\
      | ~~~~~~~~~~~~~~~~~~~~~~~
   37 | }";
      | ~~
</code>
Il finit la compilation et upload sur le 8266 mais le programme ne marche pas, voici le genre de message que j'obtiens:
<code>5:06:04.040 -> Test frequency : 433.760010
15:06:07.224 -> 
15:06:07.224 -> --------------- CUT HERE FOR EXCEPTION DECODER ---------------
15:06:07.224 -> 
15:06:07.224 -> Soft WDT reset
15:06:07.224 -> 
15:06:07.224 -> Exception (4):
15:06:07.224 -> epc1=0x4010062e epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000000 depc=0x00000000
15:06:07.224 -> 
15:06:07.224 -> >>>stack>>>
</code>
Je n'arrive pas à le décoder, j'ai un problème avec mon EspExceptionDecoder

Une idée ? Merci

**Comment by babyloon00** [comment]
J'ai pu décoder le stack.
C'est un problème dans la class SPI
J'ai essayé de recopier avec la version 3,0,2 de ESP8266 mais le problème persiste

<code>
x4020e3e3: SPIClass::beginTransaction(SPISettings) at /Users/damien/Library/Arduino15/packages/esp8266/hardware/esp8266/3.0.2/libraries/SPI/SPI.cpp line 145
0x401005e0: __digitalWrite(uint8_t, uint8_t) at /Users/damien/Library/Arduino15/packages/esp8266/hardware/esp8266/3.0.2/cores/esp8266/core_esp8266_wiring_digital.cpp line 87
0x40201050: wiringPiSPIDataRW(int, unsigned char*, int) at /Users/damien/Downloads/watermeter2mqtt/cc1101.cpp line 116
0x4020106f: wiringPiSPIDataRW(int, unsigned char*, int) at /Users/damien/Downloads/watermeter2mqtt/cc1101.cpp line 123</code>
